### ISSUE TITLE
Infinite recursion on StartError

### ISSUE THREAD

**Issue by DomWilliams0** [AlexanderThaller/prometheus_exporter#4]
[The `Display` implementation for `StartError`](https://github.com/AlexanderThaller/prometheus_exporter/blob/master/src/lib.rs#L53) recurses infinitely, I assume it's supposed to reuse the `Debug` implementation instead?

```rust
impl fmt::Display for StartError {
    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
        write!(f, "{}", self)
    }
}
```

Thanks for the library by the way, a very easy drop-in :grin:

**Comment by AlexanderThaller** [comment]
It should be fixed now. I need to put some love into this project. Thank you for bringing this up!

**Comment by AlexanderThaller** [comment]
I published the fixed version as 0.6. Gonna close the issue. Thank you again!

**Comment by DomWilliams0** [comment]
Confirmed it no longer loops forever in 0.6, cheers!
